modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -430,6 +430,12 @@ public ObjectEntry addObjectEntry(
 		finally {
 			ObjectActionThreadLocal.setClearObjectEntryIdsMap(
 				clearObjectEntryIdsMap);
+
+			for (Long dlFileEntryId : _dlFileEntryIds) {
+				TempFileEntryUtil.deleteTempFileEntry(dlFileEntryId);
+			}
+
+			_dlFileEntryIds.clear();
```

Comment:
@vngoliveira you are deleting temporary files only when adding a new object entry (`addObjectEntry`). 

What if I update an object entry by uploading a new temporary file?